### PR TITLE
add provider named release in dicehub

### PR DESCRIPTION
**Pull request by ZLValien**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

add provider named release in dicehub

### CODE REVIEW

**Comment by ZLValien** [comment]
I think GetReleaseNamesByApp and GetAppIDsByProjectAndVersion can't return right data。
{Path: "/api/releases/actions/get-name", Method: http.MethodGet, Handler: e.ListReleaseName}
{Path: "/api/releases/actions/get-latest", Method: http.MethodGet, Handler: e.GetLatestReleases}
These two interfaces use them, can I delete them?

**Comment by sfwn** [comment]
> I think GetReleaseNamesByApp and GetAppIDsByProjectAndVersion can't return right data。
> {Path: "/api/releases/actions/get-name", Method: http.MethodGet, Handler: e.ListReleaseName}
> {Path: "/api/releases/actions/get-latest", Method: http.MethodGet, Handler: e.GetLatestReleases}
> These two interfaces use them, can I delete them?

find invoker and ask UI team.

**Comment by ZLValien** [comment]
> > I think GetReleaseNamesByApp and GetAppIDsByProjectAndVersion can't return right data。
> > {Path: "/api/releases/actions/get-name", Method: http.MethodGet, Handler: e.ListReleaseName}
> > {Path: "/api/releases/actions/get-latest", Method: http.MethodGet, Handler: e.GetLatestReleases}
> > These two interfaces use them, can I delete them?
> 
> find invoker and ask UI team.

I can submit to delete their PR after this PR, so they will be retained in this PR

File: modules/dicehub/service/release/release.go [hunk 1]
```diff
@@ -313,7 +313,7 @@ func (r *Release) GetIosPlist(orgID int64, releaseID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	if orgID != 0 && release.OrgID != orgID { // 内部调用时，orgID为0
+	if orgID != 0 && release.OrgID != orgID { // 调用时内部，orgID为0
```
Comment (sfwn) [inline modules/dicehub/service/release/release.go:316]: typo

File: pkg/common/apis/header.go [hunk 2]
```diff
@@ -56,6 +56,16 @@ func HTTPLanguage(r *http.Request) i18n.LanguageCodes {
 	return langs
 }
 
+// GetApplicationId .
+func GetApplicationId(ctx context.Context) string {
```
Comment (sfwn) [inline pkg/common/apis/header.go:60]: don't put this here. It's not common.
`applicationId` or `applicationID` or `application-id` ?

File: pkg/common/apis/header.go [hunk 3]
```diff
@@ -56,6 +56,16 @@ func HTTPLanguage(r *http.Request) i18n.LanguageCodes {
...
+	return GetHeader(ctx, "applicationId")
+}
+
+// GetProjectID .
+func GetProjectID(ctx context.Context) string {
```
Comment (sfwn) [inline pkg/common/apis/header.go:65]: remove it.

File: pkg/common/apis/header.go [hunk 4]
```diff
@@ -76,6 +86,11 @@ func GetIntUserID(ctx context.Context) (int64, error) {
 	return strconv.ParseInt(GetUserID(ctx), 10, 64)
 }
 
+// GetVersion .
+func GetVersion(ctx context.Context) string {
```
Comment (sfwn) [inline pkg/common/apis/header.go:90]: remove it.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dicehub/service/release/release.go (hunk 1 above)
Comment (ZLValien) [reply modules/dicehub/service/release/release.go]: done

File: pkg/common/apis/header.go (hunk 2 above)
Comment (ZLValien) [reply pkg/common/apis/header.go]: Have moved this to release

File: pkg/common/apis/header.go (hunk 3 above)
Comment (ZLValien) [reply pkg/common/apis/header.go]: done

File: pkg/common/apis/header.go (hunk 4 above)
Comment (ZLValien) [reply pkg/common/apis/header.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**